src/diffusers/modeling_utils.py

```diff
@@ -466,7 +419,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             # if device_map is Non,e load the state dict on move the params from meta device to the cpu
             if device_map is None:
                 param_device = "cpu"
-                state_dict = load_state_dict(model_file)
+                state_dict = load_state_dict(model_file, is_pytorch)
```

Comment:
Actually could we not use a flag here at all? And instead decide in `load_state_dict` depending on the model file name (it has to end with either `SAFETENSORS_WEIGHTS_NAME` or `WEIGHTS_NAME` whether it's `is_pytorch` or not. It's a bit confusing to me to pass around a `is_pytorch` flag here

Reply:
Again I think the behavior should not on the filename, but on what you did previously.

It's bikeshedding at this point, I'll try to find something even more elegant.